### ISSUE TITLE
`TextFormField` throws `_CastError`

### ISSUE THREAD

**Issue by untp** [flutter/flutter#89913]
## Steps to Reproduce

Create a new project and edit main.dart as:
```dart
import 'package:flutter/material.dart';

void main() =>
    runApp(MaterialApp(home: Scaffold(body: Center(child: TextFormField()))));
```
Then tap on TextFormField and tap any letter on keyboard.

**Expected results:**
No exceptions.

**Actual results:**
```
══╡ EXCEPTION CAUGHT BY SERVICES LIBRARY ╞══════════════════════════════════════════════════════════
The following _CastError was thrown during a platform message callback:
Null check operator used on a null value

When the exception was thrown, this was the stack:
#0      KeyEventManager._convertRawEventAndStore (package:flutter/src/services/hardware_keyboard.dart:892:59)
#1      RawKeyboard.handleRawKeyEvent (package:flutter/src/services/raw_keyboard.dart:677:17)
#2      KeyEventManager.handleRawKeyMessage (package:flutter/src/services/hardware_keyboard.dart:818:33)
#3      BasicMessageChannel.setMessageHandler.<anonymous closure>
(package:flutter/src/services/platform_channel.dart:73:49)
#4      BasicMessageChannel.setMessageHandler.<anonymous closure>
(package:flutter/src/services/platform_channel.dart:72:47)
#5      _DefaultBinaryMessenger.setMessageHandler.<anonymous closure> (package:flutter/src/services/binding.dart:379:35)
#6      _DefaultBinaryMessenger.setMessageHandler.<anonymous closure> (package:flutter/src/services/binding.dart:376:46)
#7      _invoke2.<anonymous closure> (dart:ui/hooks.dart:205:15)
#11     _invoke2 (dart:ui/hooks.dart:204:10)
#12     _ChannelCallbackRecord.invoke (dart:ui/channel_buffers.dart:42:5)
#13     _Channel.push (dart:ui/channel_buffers.dart:132:31)
#14     ChannelBuffers.push (dart:ui/channel_buffers.dart:329:17)
#15     PlatformDispatcher._dispatchPlatformMessage (dart:ui/platform_dispatcher.dart:544:22)
#16     _dispatchPlatformMessage (dart:ui/hooks.dart:92:31)
(elided 3 frames from dart:async)
════════════════════════════════════════════════════════════════════════════════════════════════════
```

<details>
  <summary>Logs</summary>

```
[✓] Flutter (Channel stable, 2.5.0, on Microsoft Windows [Version 10.0.19041.1165], locale en-US)
    • Flutter version 2.5.0 at C:\Users\U\AppData\Local\Programs\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4cc385b4b8 (3 days ago), 2021-09-07 23:01:49 -0700
    • Engine revision f0826da7ef
    • Dart version 2.14.0

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:\Users\U\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[✗] Chrome - develop for the web (Cannot find Chrome executable at .\Google\Chrome\Application\chrome.exe)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.10.3)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.10.31424.327
    • Windows 10 SDK version 10.0.19041.0

[✓] Android Studio (version 4.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] VS Code (version 1.60.0)
    • VS Code at C:\Users\U\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.26.0

[✓] Connected device (3 available)
    • SM A505F (mobile) • [redacted]  • android-arm64  • Android 11 (API 30)
    • Windows (desktop) • windows     • windows-x64    • Microsoft Windows [Version 10.0.19041.1165]
    • Edge (web)        • edge        • web-javascript • Microsoft Edge 93.0.961.44

! Doctor found issues in 1 category.
```

</details>

**Comment by untp** [comment]
Can't reproduce on latest master.
```
[✓] Flutter (Channel master, 2.6.0-1.0.pre.254, on Microsoft Windows [Version 10.0.19041.1165], locale en-US)
    • Flutter version 2.6.0-1.0.pre.254 at C:\Users\U\AppData\Local\Programs\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision e0b56dbf48 (7 hours ago), 2021-09-11 00:52:02 -0400
    • Engine revision abb1980f65
    • Dart version 2.15.0 (build 2.15.0-96.0.dev)
```
| `stable` | `master` |
| - | - |
| ⚠️ Issue reproduced | ✅ No Issue |

**Comment by untp** [comment]
Can we hotfix this issue to stable channel?

**Comment by danagbemava-nc** [comment]
Hi @untp, do you experience this issue with any other device apart from the one specified in your `flutter doctor -v`?

**Comment by untp** [comment]
@danagbemava-nc I tested on windows there is no issue. The issue is reproduced on some android devices.

I think it is not important to reproduce this issue. Because the issue is already fixed by #88967. I just want to #88967 hotfixed to stable channel.
